api/models/site.js

```diff
@@ -42,6 +68,16 @@ function toJSON() {
   object.createdAt = object.createdAt.toISOString();
   object.updatedAt = object.updatedAt.toISOString();
 
+  // prevent any invalid domain or demoDomain URLs from being returned
+  // or used in the creation of viewLink or demoViewLink
+  if (!isEmptyOrUrl(object.domain, false)) {
+    object.domain = null;
+  }
+
+  if (!isEmptyOrUrl(object.demoDomain, false)) {
```

Comment:
How about rewriting existing URLs in-place within the database to normalize them all to what you want to enforce going forward? Then you don't need special-case logic for it.